ocaml/xapi/xapi_pci_helpers.ml

```diff
@@ -103,33 +98,31 @@ let get_host_pcis () =
             d.vendor_id d.device_id d.device_class;
           let vendor = { id = d.vendor_id
                        ; name = lookup_vendor_name access d.vendor_id
```

Comment:
how did `lookup_vendor_name` change from returning an `t option` to returning a `t`? was it previously from `pciutil` but now from `pci`?

could this change cause unexpected exceptions?

Reply:
> how did lookup_vendor_name change from returning an t option to returning a t? was it previously from pciutil but now from pci?

This is an interface change between the versioned and the upstream pci versions, as explained in the commit:
```
The interface changes because the vendored library started using Option
with the intention to fix crashes. After the fact it was seen what the
actual solution to the crashes was. The proper solution was upstreamed
and the interface change was not rolled back in xapi, until now.
```

> i assume it returns the empty string instead of None now?

Now it return `Vendor <provided id>` for example `Vendor 00ffef`.